### PR TITLE
feat(server): warning for non web target (#2026)

### PR DESCRIPTION
**Pull request by Kristonitas**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Modified existing tests inside `addEntries.test.js`.

### Motivation / Use-Case

Help new developers (who check terminal output) to more quickly detect potential misconfiguration when developing a web app while setting `target: "node"`. A more detailed explanation of this: https://github.com/webpack/webpack-dev-server/issues/2026#issuecomment-508679342. I acknowledge this is something very obvious once you know about the configuration.

### Breaking Changes

If anyone is doing strict validation of `webpack-dev-server` output and using the configuration in question, this might affect them. It should be quite easy to filter/ignore or modify.

### Additional Info

I made this minimal repo with two scripts to see the different behaviour with `"node"` and `"web"` targets: https://github.com/Kristonitas/live-reload-fail

### CODE REVIEW

**Comment by anikethsaha** [comment]
@Kristonitas the CLA is not signed, please sign it.

if its already signed and still showing not signed, close the PR and reopen it !

**Comment by Kristonitas** [comment]
@evilebottnawi 
Can you comment on this PR?
If you think I should take another approach or if this is not worth being added let me know.

File: lib/utils/addEntries.js [hunk 1]
```diff
@@ -82,6 +82,12 @@ function addEntries(config, options, server) {
       )
         ? [clientEntry]
         : [];
+      if (!webTarget) {
+        log.warn(`Non web target selected in dev server config`);
+        if (config.liveReload !== false) {
+          log.error(`Live reload won't work with non web target`);
+        }
+      }
```
Comment (evilebottnawi) [inline lib/utils/addEntries.js]: Let's use `info` for this, warn may annoy devepoers

**Comment by alexander-akait** [comment]
Close in favor https://github.com/webpack/webpack-dev-server/pull/3094, sorry for delay :disappointed:

**Pull request closed without merging**